### DISCUSSION TITLE
dds on linux  x86 compile error

### DISCUSSION THREAD

**Discussion by kait1234** [eProsima/Fast-DDS#5328] [Q&A]
I using cmake compile source, https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html ,in this file 3.1.4. CMake installation.
```
_cd ~/Fast-DDS
git clone https://github.com/eProsima/Fast-DDS.git
mkdir Fast-DDS/build
cd Fast-DDS/build
cmake ..  -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install_
```
then ,error showing:
**[ 61%] Building CXX object src/cpp/CMakeFiles/fastdds.dir/rtps/network/NetworkFactory.cpp.o
In file included from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/transport/TCPChannelResourceBasic.h:20:0,
                 from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/transport/TCPAcceptorBasic.h:19,
                 from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/transport/TCPTransportInterface.h:34,
                 from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/network/NetworkFactory.cpp:27:
/root/Fast-DDS/Fast-DDS/src/cpp/rtps/transport/TCPChannelResource.h:279:45: 错误：模板参数数目不对(不应是 1 个而应是 2 个)
             asio::basic_socket<asio::ip::tcp>& socket,
                                             ^
In file included from /usr/include/asio/basic_datagram_socket.hpp:20:0,
                 from /usr/include/asio.hpp:19,
                 from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/transport/TCPTransportInterface.h:23,
                 from /root/Fast-DDS/Fast-DDS/src/cpp/rtps/network/NetworkFactory.cpp:27:
/usr/include/asio/basic_socket.hpp:41:7: 附注：provided for ‘template<class Protocol, class SocketService> class asio::basic_socket’**
how  to do？thanks！

**Comment by JesusPoderoso** [comment]
Closing as duplicated from:
- #5371
